backend/LeetWars.Core/LeetWars.Core.BLL/Services/UserService.cs

```diff
@@ -181,16 +192,106 @@ public async Task<UserFullDto> UpdateUserRankAsync(EditUserDto userDto)
         return _mapper.Map<UserFullDto>(user);
     }
 
-    public async Task<List<UserDto>> GetLeaderBoardAsync(PageSettingsDto? page)
+    public async Task<List<UserDto>> GetLeaderBoardAsync(LeaderBoardPageSettingsDto page)
     {
-        var users = _context.Users.OrderByDescending(u => u.TotalScore).AsQueryable();
-        if (page is not null)
+        var query = _context.Users.OrderByDescending(u => u.TotalScore).AsQueryable();
+
+        if (page.GetFriends)
+        {
+            var currentUser = await GetCurrentUserEntityAsync();

```

Comment:
do we really need to load all information about user just for using Friendship?

Reply:
but this is the leaderboard, here we use a lot of info...

Reply:
You use only Friendships, but load a lot of other data
![image](https://github.com/BinaryStudioAcademy/bsa-2023-leetwars/assets/38536017/e72858b2-8a4e-415c-8dcd-007f2aa5ebf7)

Reply:
aa, i got it